src/DynamoCoreWpf/Utilities/CrashReportTool.cs

```diff
@@ -0,0 +1,228 @@
+
+                var cerDir = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), "DynamoCER_Report_" +
+                    DateTime.Now.ToUniversalTime().ToString("yyyy-MM-dd-HH-mm-ss")));
+
+                using (Scheduler.Disposable.Create(() => { cerDir.Delete(); }) )
+                {
+                    var filesToSend = new List<string>();
+                    if (args.SendLogFile && model != null)
+                    {
+                        string logFile = Path.Combine(cerDir.FullName, "DynamoLog.log");
+
+                        File.Copy(model.Logger.LogPath, logFile);
+                        // might be usefull to dump all loaded Packages into
+                        // the log at this point.
+                        filesToSend.Add(logFile);
+                    }
+
+                    if (args.SendSettingsFile && model != null)
+                    {
+                        string settingsFile = Path.Combine(cerDir.FullName, "DynamoSettings.xml");
+                        File.Copy(model.PathManager.PreferenceFilePath, settingsFile);
+
+                        filesToSend.Add(settingsFile);
+                    }
+
+                    if (args.SendDynFile && model != null)
+                    {
+                        var dynFilePath = Path.Combine(cerDir.FullName, "DynamoModel.dyn");
+                        model.CurrentWorkspace.Save(dynFilePath);
+
+                        filesToSend.Add(dynFilePath);
+                    }
+
+                    if (args.SendRecordedCommands && args.viewModel != null)
+                    {
+                        filesToSend.Add(args.viewModel.DumpRecordedCommands());
+                    }
+
+                    var extras = string.Join(" ", filesToSend.Select(f => "/EXTRA " + f));
+
+                    string appConfig = "";
+                    if (model != null)
+                    {
+                        var appName = string.IsNullOrEmpty(model.HostAnalyticsInfo.HostName) ? Process.GetCurrentProcess().ProcessName :
+                            model.HostAnalyticsInfo.HostName;
+                        appConfig = $@"<ProductInformation name=\""{appName}\"" build_version=\""{model.Version}\"" " +
+                        $@"registry_version=\""{model.Version}\"" registry_localeID=\""{CultureInfo.CurrentCulture.LCID}\"" uptime=\""0\"" " +
+                        $@"session_start_count=\""0\"" session_clean_close_count=\""0\"" current_session_length=\""0\"" />";
+                    }
+
+                    var miniDumpFilePath = CreateMiniDumpFile(cerDir.FullName);
+                    var upiConfigFilePath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "upiconfig.xml");
+
+                    var cerArgs = $"/UPITOKEN {upiConfigFilePath} /DMP {miniDumpFilePath} /APPXML \"{appConfig}\" {extras}";
+                    Process.Start(new ProcessStartInfo(CerToolLocation, cerArgs)).WaitForExit();
```

Comment:
is there any reason to WaitForExit()? Can't Dynamo just close and CER will go off and do its work?

Reply:
Yes it can. I wanted CER to behave the same as our inhouse crash report mechanism (which is modal)